### PR TITLE
Improve resilience of app

### PR DESCRIPTION
**Pull request by npentrel**
Hey Danny, I played around with the code and found some bugs - I'll comment on the code to show what I've done. 

Nice app btw, I will add some of it to the docs and link to it :)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,16 +1,14 @@
 import app
 import time
-import asyncio
```
Comment (npentrel) [inline app.py:3]: unnecessary imports removed

File: app.py [hunk 2]
```diff
@@ -32,18 +30,27 @@ def __init__(self):
         else:
             self.drv = None
             self.enable_pin = None
+
         eventbus.on(ButtonDownEvent, self._handle_buttondown, self)
-        eventbus.on_async(HexpansionInsertionEvent, self.handle_hexpansion_insertion, self)
-        eventbus.on_async(HexpansionRemovalEvent, self.handle_hexpansion_removal, self)
+        eventbus.on(HexpansionInsertionEvent, self.handle_hexpansion_insertion, self)
```
Comment (npentrel) [inline app.py:35]: `eventbus.on` goes with synchronous methods (which you used). `eventbus.on_async` would require you to declare `handle_hexpansion_insertion` as an async method. They fail if you pass the wrong type.

File: app.py [hunk 3]
```diff
@@ -32,18 +30,27 @@ def __init__(self):
...
+        eventbus.on(HexpansionRemovalEvent, self.handle_hexpansion_removal, self)
 
 
-    def handle_hexpansion_insertion(self):
+    def handle_hexpansion_insertion(self, event):
```
Comment (npentrel) [inline app.py:39]: Eventbus passes in 2 arguments so this fails if you don't add this

File: app.py [hunk 4]
```diff
@@ -32,18 +30,27 @@ def __init__(self):
...
         self.hexpansion_config = self.scan_for_hexpansion()
+        self.last_jitter_time = time.ticks_ms()
```
Comment (npentrel) [inline app.py:41]: If you remove and insert the hexpansion, you need to reset these, especially cuz people could be inserting the hexpansions in a nother slot.

File: app.py [hunk 5]
```diff
@@ -65,36 +72,37 @@ def scan_for_hexpansion(self): # Is there a proper way of doing this so I don't
...
 
-
-
     def _cleanup(self):
         eventbus.remove(ButtonDownEvent, self._handle_buttondown, self.app)
 
     def update(self, delta):
         if self.hexpansion_config is not None:
-            self.jitter_randomly()
+            try:
```
Comment (npentrel) [inline app.py:102]: Wrapping this in a try catch to avoid the app crashing if you remove the hexpansion while it's jittering

**Comment by walkerdanny** [comment]
Looks good, thanks!

**Pull request merged**